src/components/Match/VisionMap.jsx

```diff
@@ -25,44 +41,111 @@ const senWard = (style, iconSize) => (<svg style={style} width={iconSize} height
   </defs>
 </svg>);
 
-export default function VisionMap({
+// TODO Hero icon on ward circles?
+class VisionMap extends React.Component {
+  componentWillMount() {
+    this.setState({
+      obsIcons: [],
+      senIcons: [],
+      enabledIndex: {},
+    });
+  }
+  super() {
+    this.updateMap = this.updateMap.bind(this);
+  }
+  updateMap(event, checked, index) {
+    const newEnabledIndex = Object.assign({}, this.state.enabledIndex, {
+      [index]: checked,
+    });
+    this.setState(Object.assign({}, this.state, {
+      enabledIndex: newEnabledIndex,
+    }));
+  }
+  render() {
+    const match = this.props.match;
+    const width = this.props.width;
+    const enabledIndex = this.state.enabledIndex;
+    const iconSize = width / 12;
+    const style = ward => ({
+      position: 'absolute',
+      top: ((width / 127) * ward.y) - (iconSize / 2),
+      left: ((width / 127) * ward.x) - (iconSize / 2),
+    });
+    const obsIcons = [];
+    const senIcons = [];
+    Object.keys(enabledIndex).forEach((index) => {
+      if (enabledIndex[index]) {
+        if (match && match.players && match.players[index]) {
+          const obs = (match.players[index].posData && match.players[index].posData.obs) || [];
+          const sen = (match.players[index].posData && match.players[index].posData.sen) || [];
+          const stroke = isRadiant(match.players[index].player_slot) ? 'green' : 'red';
+          obs.forEach(ward => obsIcons.push(obsWard(style(ward), stroke, iconSize)));
+          sen.forEach(ward => senIcons.push(senWard(style(ward), stroke, iconSize)));
+        }
+      }
+    });
+    return (<Row>
+      <Col md={6}>
+        <div
+          style={{
+            position: 'relative',
+            top: 0,
+            left: 0,
+            width: this.props.width,
+          }}
+        >
+          <img width={this.props.width} src="/assets/images/map.png" role="presentation" />
+          {obsIcons}
+          {senIcons}
+        </div>
+      </Col>
+      <Col md={6}>
+        {
+        /*
```

Comment:
old style selector.  I think the new one is better, but keeping this around for easy swapping until we finalize.